woodwork/type_sys/inference_functions.py

```diff
@@ -17,26 +17,22 @@ def get_inference_sample(series: pd.Series) -> pd.Series:
 
 
 def categorical_func(series):
-    categorical_threshold = ww.config.get_option('categorical_threshold')
-    numeric_categorical_threshold = ww.config.get_option('numeric_categorical_threshold')
+    if pdtypes.is_categorical_dtype(series.dtype):
+        return True
```

Comment:
> Do we have specific tests that confirm that he category dtype will end up with Categorical logical type in situations where Categorical is not inferred otherwise?

Not sure.  I suppose that whole edge case depends on behavior defined in the calling context of `categorical_func` which is `TypeSystem.infer_logical_type`.  Are you saying that you think we need a test to touch the edge case of when a series has the categorical dtype but might possibly still be inferred as some other type by the logic in `TypeSystem.infer_logical_type`?

Reply:
I'm more thinking about the opposite situation, I think, in which a column is below the categorical threshold but has the category dtype so it's still `Categorical` logical type. Just would want to confirm that behavior. 

But I think the edge case you brought up is also relevant, though it might be maybe something like a column of emails with dtype category. If the email inference recognizes the email values, then the logical type should be email since it's a more specific type. But that's more of a type system test than a ltype inference test.

Reply:
> But that's more of a type system test than a ltype inference test.

Yup, agreed.

> a column is below the categorical threshold but has the category dtype so it's still Categorical logical type

Sure, I guess we can test for that if it seems like an important property we want our system to have.  And, just to confirm, this would be the correct behavior, right?  If it's explicitly marked as being categorical by Pandas (by possessing the categorical dtype), then it should also have the Categorical logical type?

Reply:
(realize I meant _above_ the categorical threshold in the comment above, which reminds me that we should update the `setting_config_options` guide as well here).

I think this behavior makes sense, though probably worth getting input from @gsheni  and @thehomebrewnerd to see if they agree. Though, it's worth noting that there will not be an equivalent in Koalas, which has no `category` dtype, so the `string` dtype gets used.

Reply:
> realize I meant *above* the categorical threshold in the comment above

Yep, assumed that's what you meant.

Reply:
I think it makes sense that a column with a `category` dtype would get inferred as a `Categorical` column in Woodwork, even if its values would cause it to be inferred as something else based on the threshold approach.

Reply:
Recommended change made in this commit: https://github.com/alteryx/woodwork/pull/1065/commits/cd3456a8e589637f4fede17fd72c9934679d1a78